scaffolding/src/commands.rs

```diff
@@ -32,7 +30,10 @@ impl CommandApplicator {
             match self.rx.try_recv() {
                 Ok(cmd)             => {
                     redraw = true;
-                    cmd.apply(&mut terminal).unwrap_or_else(exit_on_io_error);
+                    cmd.apply(&mut terminal).unwrap_or_else(|e| {
+                        panic!("Error applying command '{}' due to error: {}",
+                               cmd.repr(), e);
```

Comment:
> On this branch, when I enter ctrl+D, the terminal does not immediately exist, and when I enter another character, I hit the panic that you reintroduced at scaffolding/src/commands.rs:34.

I removed the `exit_on_io_error` that was in use here mostly because of the message, "Exiting process due to tty error". It's not clear that the only possible error here is a "tty error", but you are right this should probably continue to terminate the GTK main process (assuming we only ever instantiate one Terminal). I also thought it would be a good opportunity to prompt discussion about why the code that was here won't work as written moving forward.

Specifically, it is worth pointing out that that approach will not be sustainable as we begin instantiating multiple Terminals--one for each tab in the gtk app. In fact, it's probably best for `CommandApplicator::apply()` to return a `Result` so that the behavior on IO error can be handled in the Terminal widget (which I expect to be an encapsulation of the Terminal and DrawingArea constructs). 

Based on your description of the behavior of `scaffolding` after you send the EOF character I am guessing the problem in that specific case is that the next character you send cannot be written to the pty file descriptor--if I somehow reached this code path on a Linux box it would probably be EIO similar to what happens when I try reading the closed file descriptor.